### PR TITLE
Med 54 local doctor profile

### PR DESCRIPTION
**Pull request by skazzp**
**Profile page:**

- Added Profile Page UI
- Added Profile form validation

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

**Profile page general view:**
image

### CODE REVIEW

File: apps/medon-fe/src/components/ProfileForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,285 @@
+import { useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { DATE_FORMAT_REG } from 'utils/constants/dateFormat';
+import dayjs from 'dayjs';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Spin } from 'antd';
+import { useTranslation } from 'react-i18next';
+import { ROLES, ROLE_OPTIONS, SPECIALITY_OPTIONS } from 'utils/constants/roles';
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/ProfileForm/index.tsx:11]: import order

node_modules
absolute path

File: apps/medon-fe/src/components/ProfileForm/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    cursor: default !important;
+  }
+`;
+
+export const ImageContainer = styled.div`
+  display: flex;
+  margin-bottom: 2rem;
+`;
+
+export const StyledInput = styled(Input)``;
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/components/ProfileForm/styles.tsx]: why create StyledInput without style

File: apps/medon-fe/src/app/app.tsx [hunk 3]
```diff
@@ -15,6 +16,7 @@ function App() {
       <Route path="/forget-password" element={<ForgetPassword />} />
       <Route path="/reset-password" element={<ResetPassword />} />
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/app/app.tsx]: delete reset password route without token

File: apps/medon-fe/src/styled.d.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import 'styled-components';
 declare module 'styled-components' {
   export interface DefaultTheme {
     colors: {
+      BACKGROUND_PRIMARY: string;
```
Comment (Mati-Pereira) [inline apps/medon-fe/src/styled.d.ts]: constansts should be lowercase

File: apps/medon-fe/src/components/ProfileForm/index.tsx (hunk 1 above)
Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/index.tsx:11]: fixed

File: apps/medon-fe/src/components/ProfileForm/styles.tsx (hunk 2 above)
Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/styles.tsx]: deleted

File: apps/medon-fe/src/app/app.tsx (hunk 3 above)
Comment (skazzp) [reply apps/medon-fe/src/app/app.tsx]: done

File: apps/medon-fe/src/styled.d.ts (hunk 4 above)
Comment (skazzp) [reply apps/medon-fe/src/styled.d.ts]: fixed

**Review by Mati-Pereira — APPROVED** [review]
awesome

File: apps/medon-fe/src/components/ProfileForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            )}
+            <StyledButton size="large" htmlType="submit" disabled={true}>
+              {t('profileForm.changePasswordBtn')}
+            </StyledButton>
+          </ButtonContainer>
+        </>
+      )}
+    </Container>
+  );
+}
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/index.tsx:195]: component is to big, split, logic move to hooks, create reusable component where possible or just smaller one

Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/index.tsx:195]: moved all inputs in separate components, no logic in component

File: apps/medon-fe/src/app/app.tsx [hunk 6]
```diff
@@ -6,15 +6,16 @@ import ResetPassword from 'pages/ResetPassword';
 import ResendConfirmation from 'pages/ResendConfirmation';
 import UpdatePassword from 'pages/UpdatePassword';
 import Navigation from 'components/Navigation/index';
+import ProfilePage from 'pages/ProfilePage';
```
Comment (lyudad) [inline apps/medon-fe/src/app/app.tsx:9]: import ResendConfirmation from 'pages/ResendConfirmation';
import UpdatePassword from 'pages/UpdatePassword';
import ProfilePage from 'pages/ProfilePage';

import Navigation from 'components/Navigation/index';

File: apps/medon-fe/src/app/app.tsx [hunk 7]
```diff
@@ -6,15 +6,16 @@ import ResetPassword from 'pages/ResetPassword';
...
 
 function App() {
   return (
     <Routes>
       <Route path="/login" element={<Login />} />
       <Route path="/register" element={<RegistrationPage />} />
       <Route path="/forget-password" element={<ForgetPassword />} />
-      <Route path="/reset-password" element={<ResetPassword />} />
       <Route path="/dashboard" element={<Navigation />} />
+      <Route path="/profile" element={<ProfilePage />} />
```
Comment (lyudad) [inline apps/medon-fe/src/app/app.tsx]: move all path to constant file

File: apps/medon-fe/src/components/ProfileForm/ProfileInput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+            status={error ? 'error' : undefined}
+            placeholder={`${t('profileForm.firstName.placeholder')}`}
+            {...field}
+          />
+        )}
+      />
+      {error && <ErrorMsg role="alert">{t(`${error}`)}</ErrorMsg>}
+    </>
+  );
+}
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/ProfileInput/index.tsx]: what a difference between ProfileInput and not profile input, please try to make components reusable, it should be just input and everyone should use it

File: apps/medon-fe/src/components/ProfileForm/ProfileSelect/index.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            }
+            {...field}
+            options={options}
+          />
+        )}
+      />
+      {error && <ErrorMsg role="alert">{t(`${error}`)}</ErrorMsg>}
+    </>
+  );
+}
```
Comment (lyudad) [inline apps/medon-fe/src/components/ProfileForm/ProfileSelect/index.tsx]: same as input, why not to creat select for full project not just for Profile

Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/ProfileSelect/index.tsx]: done

File: apps/medon-fe/src/components/ProfileForm/ProfileInput/index.tsx (hunk 8 above)
Comment (skazzp) [reply apps/medon-fe/src/components/ProfileForm/ProfileInput/index.tsx]: created fully reusable input/select/datepicker

File: apps/medon-fe/src/app/app.tsx (hunk 6 above)
Comment (skazzp) [reply apps/medon-fe/src/app/app.tsx:9]: fixed

File: apps/medon-fe/src/app/app.tsx (hunk 7 above)
Comment (skazzp) [reply apps/medon-fe/src/app/app.tsx]: done

**Review by lyudad — APPROVED** [review]

**Pull request merged**